lib_test/test_runner.ml

```diff
@@ -422,6 +422,8 @@ let test_with_change = [
     let urn = Uri.of_string "urn:uuid:f81d4fae-7dec-11d0-a765-00a0c91e6bf6" in
     let urn2= Uri.with_scheme urn (Some "urn") in
     assert_equal ~printer urn urn2;
+    let git = Uri.of_string "git@github.com:owner/repo" in
+    assert_equal ~printer:Fmt.(to_to_string (option string)) (Uri.scheme git) None ;
```

Comment:
The path of `user@host:50/repo` should be `50/repo`. Can you add a test to make sure it isn't picked as the port number ?

Reply:
That would indeed allow me to use it in dune-release !

Reply:
All information move as a `path` part of the given `Uri.t`. Again, this PR does not want to fix your initial issue but it wants to respect a little better the RFC (and disallow `@` as a possible character of the _scheme_ part). If you really want to accept and understand something like `git@github.com:owner/repo`, you should take a look on [`emile`](https://github.com/dinosaure/emile) - with a split on `:` - which one best fits your case.